lazyflow/operators/opCompressedUserLabelArray.py

```diff
@@ -247,12 +262,24 @@ def _executeProjection2D(self, roi, destination):
             # Compute slicing within the deep array and slicing within this block
             deep_relative_intersection = numpy.subtract(intersecting_roi, input_roi.start)
             block_relative_intersection = numpy.subtract(intersecting_roi, block_start)
-                        
-            deep_data = self._getBlockDataset( entire_block_roi )[roiToSlice(*block_relative_intersection)]
+            block_relative_intersection_slicing = roiToSlice(*block_relative_intersection)
+
+            block = self._getBlockDataset( entire_block_roi )
+            deep_data = None
+            if self.Output.meta.has_mask:
+                deep_data = numpy.ma.masked_array(
+                    block["data"][block_relative_intersection_slicing],
+                    mask=block["mask"][block_relative_intersection_slicing],
+                    fill_value=block["fill_value"][()],
+                    shrink=False
+                )
+            else:
+                deep_data = block[block_relative_intersection_slicing]
+
+            # make binary and convert to float (must copy)
+            deep_data_float = deep_data.astype(numpy.float32)
+            deep_data_float[deep_data_float.nonzero()] = 1
 
-            # make binary and convert to float
-            deep_data_float = numpy.where( deep_data, numpy.float32(1.0), numpy.float32(0.0) )
-            
             # multiply by slice-index
```

Comment:
I used data of shape `(100,1000,1000)`, with `deep_data.dtype == uint8`.  But in this particular case (sparse data), your change is better.

>If we decide it needs to be where, there is numpy.ma.where, which returns a masked array that could be used in a simple branch.

Wait, so... are you saying that `numpy.where()` can't be used in lazyflow any more?  Perhaps we need to define a helper function that transparently handles both cases (masked/unmasked).  Or maybe even monkey-patch `numpy.where()`?

What other ordinary numpy functions will we need to avoid now that we want to support masked arrays?